geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/AlterQueryServiceCommand.java

```diff
@@ -0,0 +1,209 @@
+          + "' options can be specified.";
+  static final String NO_ARGUMENTS_MESSAGE =
+      "No arguments were provided. No changes have been applied.";
+  static final String NO_MEMBERS_FOUND_MESSAGE = "No members found.";
+  static final String SECURITY_NOT_ENABLED_MESSAGE =
+      "Integrated security is not enabled for this distributed system. Updating the method authorizer requires integrated security to be enabled.";
+  public static final String PARTIAL_FAILURE_MESSAGE =
+      "In the event of a partial failure of this command, re-running the command or restarting failing members should restore consistency.";
+
+  @CliCommand(value = COMMAND_NAME, help = COMMAND_HELP)
+  @CliMetaData(
+      interceptor = "org.apache.geode.management.internal.cli.commands.AlterQueryServiceCommand$Interceptor")
+  @ResourceOperation(resource = ResourcePermission.Resource.CLUSTER,
+      operation = ResourcePermission.Operation.MANAGE)
+  public ResultModel execute(
+      @CliOption(key = METHOD_AUTHORIZER_NAME,
+          help = METHOD_AUTHORIZER_NAME_HELP) String methodAuthorizerName,
+      @CliOption(key = AUTHORIZER_PARAMETERS,
+          help = AUTHORIZER_PARAMETERS_HELP) String[] authorizerParameters,
+      @CliOption(key = SINGLE_AUTHORIZER_PARAMETER,
+          help = SINGLE_AUTHORIZER_PARAMETER_HELP) String singleAuthorizerParameter) {
+    ResultModel result;
+
+    QueryConfigService queryConfigService = getQueryConfigService();
+    Set<String> parametersSet = null;
+    if (methodAuthorizerName != null) {
+      if (!isSecurityEnabled()) {
+        return ResultModel.createError(SECURITY_NOT_ENABLED_MESSAGE);
+      }
+      if (Boolean.parseBoolean(
+          System.getProperty(GEMFIRE_PREFIX + "QueryService.allowUntrustedMethodInvocation"))) {
+        return ResultModel.createError(DEPRECATED_PROPERTY_ERROR);
+      }
+      if (authorizerParameters != null) {
+        parametersSet = new HashSet<>(Arrays.asList(authorizerParameters));
+      } else if (singleAuthorizerParameter != null) {
+        parametersSet = new HashSet<>(Collections.singleton(singleAuthorizerParameter));
+      }
+      populateMethodAuthorizer(methodAuthorizerName, parametersSet, queryConfigService);
+    }
+
+    Set<DistributedMember> targetMembers = findMembers(null, null);
+    if (targetMembers.size() > 0) {
+      Object[] args = new Object[] {methodAuthorizerName, parametersSet};
+      List<CliFunctionResult> functionResults =
+          executeAndGetFunctionResult(new AlterQueryServiceFunction(), args,
+              targetMembers);
+      String footer = null;
+      for (CliFunctionResult cliResult : functionResults) {
+        if (!cliResult.isSuccessful()) {
+          footer = PARTIAL_FAILURE_MESSAGE;
+          break;
+        }
+      }
+      result =
+          ResultModel.createMemberStatusResult(functionResults, null, footer, false, true);
+    } else {
```

Comment:
Personal preference, maybe flip the if to be if (targetMembers.size() == 0) { //short block of code; } else {//big block of code}

Reply:
Done!